### PR TITLE
Replace `submit` subcommand with `sync`

### LINKED ISSUE #34: Drop `submit` subcommand
The `/lc submit` subcommand is not necessary when we have the usernames of all of the participants. What we can instead do is fetch all of the accepted solutions of our participating players from the last week in order to find the most valuable, valid submission to automatically count towards their score for the season.

### PR DESCRIPTION
**Pull request by EthanThatOneKid**
Resolves #34.

### CODE REVIEW

File: api/discord_app/sub/sync.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+/**
+ * makeSyncInteractionResponse makes the interaction response for the sync subcommand.
+ */
+export function makeSyncInteractionResponse(
+  r: api.SyncResponse,
+): APIInteractionResponse {
+  return {
+    type: InteractionResponseType.ChannelMessageWithSource,
+    data: {
+      content: `Season \`${r.season.id}\` synced.\n\n${formatScores(r.season)}`,
```
Comment (EthanThatOneKid) [inline api/discord_app/sub/sync.ts:75]: Awkward-looking vertical spacing.
image
```suggestion
      content: `Season \`${r.season.id}\` synced.\n${formatScores(r.season)}`,
```

Comment (EthanThatOneKid) [inline api/discord_app/sub/sync.ts:75]: Awkward-looking vertical spacing.
image
```suggestion
      content: `Season \`${r.season.id}\` synced.\n${formatScores(r.season)}`,
```

File: api/types.ts [hunk 2]
```diff
@@ -58,6 +75,20 @@ export interface LCQuestion {
...
+
+/**
+ * Scores is a map of scores by Discord user ID.
+ */
+export interface Scores {
+  [discord_user_id: string]: number;
+}
+
 /**
  * Season is a season of the leaderboard.
```
Comment (EthanThatOneKid) [inline api/types.ts:93]: Resolve #44.

File: lib/leaderboard/denokv/denokv_leaderboard_client.ts [hunk 3]
```diff
@@ -118,122 +132,62 @@ export class DenoKvLeaderboardClient implements LeaderboardClient {
...
+        seasonID,
+      ]);
+      if (!seasonResult.value) {
+        throw new Error("Season not found");
+      }
+
+      season = seasonResult.value;
+    } else {
+      seasonResult = await this.getLatestSeasonFromKv();
+      season = seasonResult?.value
```
Comment (EthanThatOneKid) [inline lib/leaderboard/denokv/denokv_leaderboard_client.ts:157]: Set `season` to the return value of `makeEmptySeason` if `seasonResult?.value === undefined` OR the most recently synced season is not `checkDateInWeek` of `startOfWeekUTC` and `seasonResult.value.start_date`.

File: api/dailies.ts
Comment (EthanThatOneKid) [inline api/dailies.ts:1]: - Remove `/lc submit` usage instructions from embeds returned by function `makeDailyWebhookEmbeds`. 
- Add link to visualize most recently-synced scored on acmcsuf.com (#36).

Comment (EthanThatOneKid) [reply api/dailies.ts:1]: Future work: Add link to visualize most recently-synced scored on acmcsuf.com (#36).

Comment (EthanThatOneKid) [reply api/dailies.ts:1]: > Remove `/lc submit` usage instructions from embeds returned by function `makeDailyWebhookEmbeds`.

Done in <https://github.com/acmcsufoss/lc-dailies/pull/41/commits/1ae443940444ae0239f709c68819c67954861eae>.

> Future work: Add a link to visualize the most recently synced leaderboard scores on acmcsuf.com (#36).

Followed up with <https://github.com/acmcsufoss/lc-dailies/issues/36#issuecomment-1763655531>.

Comment (EthanThatOneKid) [reply api/dailies.ts:1]: Update HANDBOOK.md accordingly.

Comment (EthanThatOneKid) [reply api/dailies.ts:1]: > Update HANDBOOK.md accordingly.

Resolved in <https://github.com/acmcsufoss/lc-dailies/pull/41/commits/0ac9cbe8c4a0d1470531006a945ced385911ae9c>.

File: lib/leaderboard/denokv/denokv_leaderboard_client.ts (hunk 3 above)
Comment (EthanThatOneKid) [reply lib/leaderboard/denokv/denokv_leaderboard_client.ts]: Resolved in <https://github.com/acmcsufoss/lc-dailies/pull/41/commits/2abcf63b35feec1b5cd0a663712c2a4d0a9a6f4f>.

File: api/types.ts (hunk 2 above)
Comment (EthanThatOneKid) [reply api/types.ts:93]: Resolved in <https://github.com/acmcsufoss/lc-dailies/pull/41/commits/1b1890ffa2cdcad24ba9bfa2473e465ceae19d9c>.

File: api/discord_app/sub/sync.ts (hunk 1 above)
Comment (EthanThatOneKid) [reply api/discord_app/sub/sync.ts]: Resolved in <https://github.com/acmcsufoss/lc-dailies/pull/41/commits/bf703086d73fb14a72bcd97e5ddc52840aba03f9>.

**Review by karnikaavelumani — APPROVED** [review]
LGTM! ʕ•̫͡•ʕ•̫͡•ʔ•̫͡•ʔ•̫͡•ʕ•̫͡•ʔ•̫͡•ʕ•̫͡•ʕ•̫͡•ʔ•̫͡•ʔ

**Pull request merged**